target/linux/realtek/files-5.15/drivers/net/phy/rtl83xx-phy.c

```diff
@@ -929,17 +986,342 @@ static int rtl8380_configure_ext_rtl8218b(struct phy_device *phydev)
 	return 0;
 }
 
+/*
+ * xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+ * This part takes care about RTL8218B bringup on RTL839x devices. It will take a
+ * lot of time and polishing to get this integrated into OpenWrt but at least we
+ * have a working configuration. 
+ */
+
+typedef struct {
+    int phy;
+    int reg;
+    int val;
+    int mask;
+} patchdata;
+
+typedef struct {
+	int portstart;
+	int portend;
+	patchdata data[];
+} patchsequence;
+
+patchsequence rtl8218b_6276A_rtl8390_perchip = {
+	0 ,  0,
+	{
+	/* Serdes */
+	{ 0, 0x1e, 0x0008, 0xffff },
+	{ 0, 0x1f, 0x0405, 0xffff }, 
+	{ 0, 0x14, 0x08EC, 0xffff },
+	{ 0, 0x1f, 0x0404, 0xffff }, 
+	{ 0, 0x17, 0x5359, 0xffff }, 
+	{ 0, 0x1f, 0x0424, 0xffff }, 
+	{ 0, 0x17, 0x5359, 0xffff }, 
+	{ 0, 0x1f, 0x042C, 0xffff }, 
+	{ 0, 0x11, 0x4000, 0xffff }, 
+	{ 0, 0x12, 0x2020, 0xffff }, 
+	{ 0, 0x1f, 0x042D, 0xffff }, 
+	{ 0, 0x11, 0xC014, 0xffff },
+	/* 0x2325 0x202a */
+	{ 0, 0x1f, 0x0464, 0xffff }, 
+	/* Rsel = 10G/FF, kp1_1 = 2 */
+	{ 0, 0x15, 0x202a, 0xffff }, 
+	/* configure VGA=0x3 */
+	{ 0, 0x12, 0x1fe0, 0xffff },
+	{ 0, 0x1f, 0x042E, 0xffff },
+	{ 0, 0x11, 0x218F, 0xffff },
+	/* 0x2261 = 0x4000, Turn off RING PLL */
+	{ 0, 0x1f, 0x044C, 0xffff },
+	{ 0, 0x11, 0x4000, 0xffff },
+	{ 0, 0x1f, 0x0460, 0xffff }, 
+	{ 0, 0x10, 0x4800, 0xffff },
+	{ 0, 0x1f, 0x0462, 0xffff }, 
+	{ 0, 0x12, 0x96b3, 0xffff },
+	/*enable cmu lbw for Icp */
+	{ 0, 0x15, 0x6e58, 0xffff },
+	{ 0, 0x1f, 0x0464, 0xffff }, 
+	{ 0, 0x13, 0x3def, 0xffff }, 
+	{ 0, 0x16, 0xf072, 0xffff },
+	{ 0, 0x1f, 0x0465, 0xffff }, 
+	{ 0, 0x10, 0x4208, 0xffff }, 
+	{ 0, 0x11, 0x3a08, 0xffff }, 
+	{ 0, 0x12, 0x4007, 0xffff },
+	{ 0, 0x1f, 0x0464, 0xffff }, 
+	{ 0, 0x17, 0x84f5, 0xffff }, 
+	{ 0, 0x17, 0x04f5, 0xffff },
+	{ 0, 0x1e, 0x0008, 0xffff },
+	/* 15-11:Filter_0, 10-7:1=force, 0=calib */
+	{ 0, 0x1f, 0x0464, 0xffff }, 
+	{ 0, 0x12, 0x1800, 0xf800 },
+	{ 0, 0x12, 0x0780, 0x0780 },
+	/* 14-10:fitler_3, 9-5:filter_2, 4-0:filter_1 */
```

Comment:
Most useless comment : fitler typo'ed, and the one above is Capitalized...

Reply:
fixed somewhere between other rewrites ...